### PR TITLE
Add attribute comment and initial value support to import_df_structures

### PR DESCRIPTION
**Pull request by paxman101**
Comments are now imported from XML elements that have the "comment" attribute defined. Furthermore, initial value attributes are also imported. 

When added to the Ghidra database, initial values are appended to comments with the format  "\<comment\> (init-value: \<val\>)" For types that the Ghidra database doesn't have comments for, the comments in the XML are instead added as "descriptions" within Ghidra.

These changes still do not import any comments that are defined within the XML as their own elements/tags.

See https://github.com/DFHack/df_misc/issues/19 and https://github.com/DFHack/dfhack/issues/2489

### CODE REVIEW

File: ghidra/import_df_structures.java [hunk 1]
```diff
@@ -490,11 +499,22 @@ private static class CodeGen {
...
+			}
+
+			@Override
+			public String getComment() {
+				return this.comment;
+			}
 		}
 
-		public static class Field extends AnonNameHaver implements ILoweredData, IOwnsType, IHasTypeName {
+		public static class Field extends AnonNameHaver implements ILoweredData, IOwnsType, IHasTypeName, IHasComment, IHasInitValue {
```
Comment (ab9rf) [inline ghidra/import_df_structures.java:517]: this line is 127 characters long, makes it harder to read in code review
suggest putting the `implements` clause on the next line

Comment (paxman101) [reply ghidra/import_df_structures.java:517]: Sure thing

**Comment by paxman101** [comment]
I didn't implement the element comments because I had thought that it would be non-trivial as to which object to include them in within the Java program. However, I realize now that as per https://github.com/DFHack/df-structures/blob/a446b5c957d9106785e584d2d78760d5e40e06da/data-definition.xsd#L2-L17 and looking through the .xml files that we can just append any element comments to its parent's attribute comments if it exists.

E.g.
```XML
<ld:field ld:level="2" ld:meta="static-array" name="biome_corner" count="16" comment="0=Reference is NW, 1=Reference is N, 2=Reference is W, 3=Reference is current tile" ld:is-container="true">
	<ld:item ld:level="3" ld:meta="static-array" type-name="int8_t" count="16" ld:is-container="true"><ld:item ld:level="4" ld:meta="number" ld:subtype="int8_t" ld:bits="8"/></ld:item>
	<comment>
		All 4 corners touching get the same reference (which determines the biome),
		i.e. SE corner of the tile to the NW, SW corner of the tile to the
		N, NE corner of the tile to the W, and the NW corner of the current
		tile, as directed by the biome_corner value.
	</comment>
</ld:field>
```
Would give `biome_corner` a comment of `0=Reference is NW, 1=Reference is N, 2=Reference is W, 3=Reference is current tile | 		All 4 corners touching get the same reference (which determines the biome), i.e. SE corner of the tile to the NW, SW corner of the tile to the N, NE corner of the tile to the W, and the NW corner of the current tile, as directed by the biome_corner value.`
Where `|` could be any arbitrary separator. 

On second thought, also according to the schema, https://github.com/DFHack/df-structures/blob/a446b5c957d9106785e584d2d78760d5e40e06da/data-definition.xsd#L77-L80 , I interpret this as meaning the element comments (from SharedElements) within the BitfieldTypeDef element are meant to be comments for the flag-bit that immediately follows them. However, in this case, the parent would still be the BitfieldTypeDef and appending the comments to the BitfieldTypeDef would remove the positional meaning of which comment goes with which bitfield. I suppose we could add special logic for Bitfields. Although, just because the schema allows this, doesn't mean it is used within the XML files as such. 

But, I don't think the current XML files conform to this schema anyways (especially considering it was last updated 3 years ago). So I'm unsure of valid XML layouts. When I tried to validate codegen.out.xml against the schema in Notepad++ with the xml tools plugin, it said the schema was non-deterministic. I could help update the schema if need be.

**Comment by ab9rf** [comment]
At the moment let's leave plaintext elements in the XML out of what gets passed into ghidra. A lot of these are voluminous and as the field displayed ghidra is not very large they'd be not very useful anyway. It's more than enough to pick up on the `comment` tag values specified within the various elements.

Also for most of the element types, that element will ultimately generate some sort of content within ghidra that has a "description" field associated with, and that's where we want the comment for that element to go. Bitfields generate enum types, and the comments for each bitfield value should properly go on the corresponding value in that enum type. If there's no settable description field associated with the ghidra value that element maps to, it's fine to drop the comment.

I haven't looked at the XSD in years so it's entirely possible that there's something wrong with it.

I'll have a closer look at this tomorrow.

**Comment by ab9rf** [comment]
just noting that i have been looking at this a little, but not as much as i'd like due to other demands on my time. i do like what i'm seeing here, but i don't think i'll have a chance to give this the look it deserves before tuesday at the earliest. just wanted you to know i'm not ignoring it. thanks

**Review by ab9rf — APPROVED** [review]

**Pull request merged**